### PR TITLE
Add weaponsto backpack

### PR DESCRIPTION
**Pull request by Salbei2500**

### CODE REVIEW

File: functions/extract/fn_extractLoadoutFromConfig.sqf [hunk 1]
```diff
@@ -81,8 +81,43 @@ private _configValues = [] call CBA_fnc_hashCreate;
...
+    "addItemsToVest"
 ];
 
+private _value = [_configPath >> "addItemsToBackpack", "array", false] call  CBA_fnc_getConfigEntry;
+if (!(_value isEqualTo false)) then {
+    {
+        if (isClass (configFile >> "CfgWeapons" >> _x)) then {
+            private _muzzle = [_configPath >> _x >> "muzzle", "text", ""] call  CBA_fnc_getConfigEntry;
+            private _pointer = [_configPath >> _x >> "pointer", "text", ""] call  CBA_fnc_getConfigEntry;
+            private _scope = [_configPath >> _x >> "scope", "text", ""] call  CBA_fnc_getConfigEntry;
```
Comment (McDiod) [inline functions/extract/fn_extractLoadoutFromConfig.sqf]:
```suggestion
            private _scope = [_configPath >> _x >> "optics", "text", ""] call  CBA_fnc_getConfigEntry;
```

Let's call this "optics" to be consistent with `primaryWeaponOptics` etc.

File: functions/extract/fn_extractLoadoutFromConfig.sqf [hunk 2]
```diff
@@ -81,8 +81,43 @@ private _configValues = [] call CBA_fnc_hashCreate;
...
+        {
+            if (isClass (configFile >> _x >> _type)) exitWith {_check = false;}
+        } forEach ["CfgWeapons", "CfgAmmo", "CfgVehicles", "CfgMagazines", "CfgItems"];
+
+        if (_check) then {
+            private _weapon = [_configPath >> _x >> "weapon", "text", ""] call  CBA_fnc_getConfigEntry;
+            if (!(_weapon isEqualTo "") && (isClass (configFile >> "CfgWeapons" >> _weapon))) then {
+                private _muzzle = [_configPath >> _x >> "muzzle", "text", ""] call  CBA_fnc_getConfigEntry;
+                private _pointer = [_configPath >> _x >> "pointer", "text", ""] call  CBA_fnc_getConfigEntry;
+                private _scope = [_configPath >> _x >> "scope", "text", ""] call  CBA_fnc_getConfigEntry;
```
Comment (McDiod) [inline functions/extract/fn_extractLoadoutFromConfig.sqf]:
```suggestion
                private _scope = [_configPath >> _x >> "optics", "text", ""] call  CBA_fnc_getConfigEntry;
```

see above

File: README.md [hunk 3]
```diff
@@ -151,12 +151,39 @@ class Loadouts {
...
 };
 ```
+### Weapons in backpacks
+If you want to use a weapon in a backpack, the config has to look like the following:
+```sqf
+addItemsToBackpack[] = {"arifle_Mk20C_F", ...};
+class arifle_Mk20C_F {
+    muzzle = "";
+    pointer = "";
+    scope = "";
```
Comment (McDiod) [inline README.md]:
```suggestion
    optics = "";
```

see above

File: README.md [hunk 4]
```diff
@@ -151,12 +151,39 @@ class Loadouts {
...
+};
+```
+or
+```sqf
+addItemsToBackpack[] = {"FancySchmazyWeapon", ...};
+class FancySchmazyWeapon {
+    weapon = "arifle_Mk20C_F";
+    muzzle = "";
+    pointer = "";
+    scope = "";
```
Comment (McDiod) [inline README.md]:
```suggestion
    optics = "";
```

see above

File: functions/general/fn_normalizeContent.sqf [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    _value = 1;
+    if ([_hash, _key] call CBA_fnc_hashHasKey) then {
+        _value = _value + ([_hash, _key] call CBA_fnc_hashGet);
+    };
+    [_hash, _key, _value] call CBA_fnc_hashSet;
+};
+
+private _magazines = [] call CBA_fnc_hashCreate;
+private _contentForLoadout = [];
+private _backpackClass = (_unitLoadout select 5) select 0;
```
Comment (McDiod) [inline functions/general/fn_normalizeContent.sqf]: `_backpackClass` is unused in this function, which also means we don't need the `_unitLoadout` that is passed to it.

File: functions/general/fn_hashToUnitLoadout.sqf [hunk 6]
```diff
@@ -56,7 +49,7 @@ private _assignFromLoadoutHash = {
                 _value = [_value, _classNameToPickOrMapper] call _getFirstOfType;
             };
             if (typeName _classNameToPickOrMapper == "CODE") then {
-                _value = [_value] call _classNameToPickOrMapper;
+                _value = [_value, _unitLoadout] call _classNameToPickOrMapper;
```
Comment (McDiod) [inline functions/general/fn_hashToUnitLoadout.sqf]: See above - `_unitLoadout` is apparently not needed in fn_normalizeContent.

File: README.md [hunk 7]
```diff
@@ -151,12 +151,39 @@ class Loadouts {
...
 
+### Random selection
 To use a random selection of Uniforms, Vests, etc. it is possible to define them as an array. Example:
 ```sqf
 class AllUnits {
 	uniform[] = {"U_C_man_sport_1_F", "U_BG_Guerilla2_1"};
 };
 ```
+### Weapons in backpacks
+If you want to use a weapon in a backpack, the config has to look like the following:
```
Comment (McDiod) [inline README.md]:
```suggestion
If you want to add a weapon to a backpack, simply add the weapon's classname to `addItemsToBackpack` like you would with any other item. However, if you want the weapon to have attachments and/or loaded magazines, the config has to look like this:
```

**Review by McDiod — CHANGES_REQUESTED** [review]

**Review by McDiod — APPROVED** [review]

**Review by DerZade — APPROVED** [review]
Looks good to me :D
Nicely done @AdlerSalbei

File: README.md [hunk 8]
```diff
@@ -151,12 +151,39 @@ class Loadouts {
...
 
+### Random selection
 To use a random selection of Uniforms, Vests, etc. it is possible to define them as an array. Example:
 ```sqf
 class AllUnits {
 	uniform[] = {"U_C_man_sport_1_F", "U_BG_Guerilla2_1"};
 };
 ```
+### Weapons in backpacks
+If you want to add a weapon to a backpack, simply add the weapon's classname to `addItemsToBackpack` like you would with any other item. However, if you want the weapon to have attachments and/or loaded magazines, the config has to look like this:
```
Comment (McDiod) [inline README.md]:
```suggestion
If you want to add a weapon to a backpack, simply add the weapon's classname to `addItemsToBackpack` like you would with any other item. However, if you want the weapon to have attachments and/or loaded magazines, the config has to look like the following examples. Note that the weapon class has to be inside the same parent class as the `addItemsToBackpack` property where it is used.
```

@DerZade  so?

**Review by DerZade — APPROVED** [review]

File: functions/general/fn_normalizeContent.sqf [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+        _value = _value + ([_hash, _key] call CBA_fnc_hashGet);
+    };
+    [_hash, _key, _value] call CBA_fnc_hashSet;
+};
+
+private _magazines = [] call CBA_fnc_hashCreate;
+private _contentForLoadout = [];
+
+{
+    if ((typeName _x) == "ARRAY") then {
```
Comment (dedmen) [inline functions/general/fn_normalizeContent.sqf:25]: `isEqualType`

**Pull request merged**